E2E, the README.md file for E2E file need be updated.

E2E, the README.md file for E2E file need be updated.
- [Created an end-to-end environment for testing and development](https://github.com/confidential-containers/cloud-api-adaptor/tree/staging/libvirt)
```bash
root@peer-pod1:~/cloud-api-adaptor# kcli list kube
+-----------+---------+-----------+-----------------------------------------+
|  Cluster  |   Type  |    Plan   |                   Vms                   |
+-----------+---------+-----------+-----------------------------------------+
| peer-pods | generic | peer-pods | peer-pods-ctlplane-0,peer-pods-worker-0 |
+-----------+---------+-----------+-----------------------------------------+
root@peer-pod1:~/cloud-api-adaptor# kubectl get nodes
NAME                   STATUS   ROLES           AGE   VERSION
peer-pods-ctlplane-0   Ready    control-plane   35m   v1.26.2
peer-pods-worker-0     Ready    worker          32m   v1.26.2
root@peer-pod1:~/cloud-api-adaptor# virsh -c qemu:///system list
 Id   Name                   State
--------------------------------------
 1    peer-pods-ctlplane-0   running
 2    peer-pods-worker-0     running
```
-  Follow the https://github.com/confidential-containers/cloud-api-adaptor/tree/staging/test/e2e try to run the e2e test base on libvirt, got the test result as:
```bash
root@peer-pod1:~/cloud-api-adaptor# CLOUD_PROVIDER=libvirt make test-e2e
go test -v  -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=unknown' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=cbcc961d6721c77a1c9e0c446ddfdcc3002e2076-dirty'" -tags=aws,azure,ibmcloud,vsphere,libvirt -timeout 20m -count=1 ./test/e2e
Do setup
Deploy the Cloud API Adaptor
Install the controller manager
Wait for the cc-operator-controller-manager deployment be available
Install CoCo and cloud-api-adaptor
Wait for the pod cc-operator-daemon-install-tprdx be ready
Wait for the pod cloud-api-adaptor-daemonset-libvirt-5mrqm be ready
Wait for the kata runtimeclass be created
=== RUN   TestAws
    env.go:230: No test testFeatures provided, skipping test
--- PASS: TestAws (0.00s)
=== RUN   TestCreateSimplePodAzure
    azure_test.go:65: AZURE_SUBSCRIPTION_ID was not set
--- FAIL: TestCreateSimplePodAzure (0.00s)
=== RUN   TestCreateSimplePod
    ibmcloud_test.go:63: APIKEY was not set.
--- FAIL: TestCreateSimplePod (0.00s)
=== RUN   TestLibvirtCreateSimplePod
=== RUN   TestLibvirtCreateSimplePod/Simple_Peer_Pod
=== RUN   TestLibvirtCreateSimplePod/Simple_Peer_Pod/PodVM_is_created
--- PASS: TestLibvirtCreateSimplePod (60.04s)
    --- PASS: TestLibvirtCreateSimplePod/Simple_Peer_Pod (60.04s)
        --- PASS: TestLibvirtCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (0.00s)
FAIL
Delete the Cloud API Adaptor installation
Uninstall CoCo and cloud-api-adaptor
Uninstall the controller manager
Wait for the cc-operator-controller-manager deployment be deleted
FAIL	github.com/confidential-containers/cloud-api-adaptor/test/e2e	281.870s
FAIL
make: *** [Makefile:85: test-e2e] Error 1
```
We set the `CLOUD_PROVIDER` to libvirt, but the e2e test cases for other provider are executed too.

The expected result should looks like:
```bash
root@peer-pod1:~/cloud-api-adaptor# go test -v  -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=unknown' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=cbcc961d6721c77a1c9e0c446ddfdcc3002e2076-dirty'" -tags=libvirt -timeout 20m -count=1 ./test/e2e
Do setup
Deploy the Cloud API Adaptor
Install the controller manager
Wait for the cc-operator-controller-manager deployment be available
Install CoCo and cloud-api-adaptor
Wait for the pod cc-operator-daemon-install-2xgdq be ready
Wait for the pod cloud-api-adaptor-daemonset-libvirt-xdjhz be ready
Wait for the kata runtimeclass be created
=== RUN   TestLibvirtCreateSimplePod
=== RUN   TestLibvirtCreateSimplePod/Simple_Peer_Pod
=== RUN   TestLibvirtCreateSimplePod/Simple_Peer_Pod/PodVM_is_created
--- PASS: TestLibvirtCreateSimplePod (55.04s)
    --- PASS: TestLibvirtCreateSimplePod/Simple_Peer_Pod (55.04s)
        --- PASS: TestLibvirtCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (0.00s)
PASS
Delete the Cloud API Adaptor installation
Uninstall CoCo and cloud-api-adaptor
Uninstall the controller manager
Wait for the cc-operator-controller-manager deployment be deleted
ok  	github.com/confidential-containers/cloud-api-adaptor/test/e2e	266.871s
```
Or
```bash
root@peer-pod1:~/cloud-api-adaptor# BUILTIN_CLOUD_PROVIDERS=libvirt make test-e2e
go test -v  -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=unknown' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=cbcc961d6721c77a1c9e0c446ddfdcc3002e2076-dirty'" -tags=libvirt -timeout 20m -count=1 ./test/e2e
Do setup
Deploy the Cloud API Adaptor
Install the controller manager
Wait for the cc-operator-controller-manager deployment be available
Install CoCo and cloud-api-adaptor
Wait for the pod cc-operator-daemon-install-7vcqv be ready
Wait for the pod cloud-api-adaptor-daemonset-libvirt-6z6js be ready
Wait for the kata runtimeclass be created
=== RUN   TestLibvirtCreateSimplePod
=== RUN   TestLibvirtCreateSimplePod/Simple_Peer_Pod
=== RUN   TestLibvirtCreateSimplePod/Simple_Peer_Pod/PodVM_is_created
--- PASS: TestLibvirtCreateSimplePod (55.04s)
    --- PASS: TestLibvirtCreateSimplePod/Simple_Peer_Pod (55.04s)
        --- PASS: TestLibvirtCreateSimplePod/Simple_Peer_Pod/PodVM_is_created (0.00s)
PASS
Delete the Cloud API Adaptor installation
Uninstall CoCo and cloud-api-adaptor
Uninstall the controller manager
Wait for the cc-operator-controller-manager deployment be deleted
ok  	github.com/confidential-containers/cloud-api-adaptor/test/e2e	271.871s

```

--- a/test/e2e/README.md
+++ b/test/e2e/README.md
@@ -8,7 +8,7 @@ As long as the cloud provider support is implemented on this framework, you can 
 as shown below for *libvirt*:
 
 ```
-$ CLOUD_PROVIDER=libvirt make test-e2e
+$ BUILTIN_CLOUD_PROVIDERS=libvirt make test-e2e
 ```
 
 The above command run tests on an existing cluster. It will look for the kubeconf file exported on the
@@ -17,7 +17,7 @@ The above command run tests on an existing cluster. It will look for the kubecon
 You can instruct the tool to provision a test environment though, as shown below:
 
 ```
-$ TEST_E2E_PROVISION=yes CLOUD_PROVIDER=libvirt make test-e2e
+$ TEST_E2E_PROVISION=yes BUILTIN_CLOUD_PROVIDERS=libvirt make test-e2e
 ```
 
 Each provider must have a provisioner implementation so that the framework is able to perform operations on the cluster. The provisioner will likely to need additional information (e.g. login credentials), and those are passed via a properties file with the following format:
@@ -31,13 +31,13 @@ key2 = "value2"
 You should use the `TEST_E2E_PROVISION_FILE` variable to specify the properties file path, as shown below:
 
 ```
-$ TEST_E2E_PROVISION=yes TEST_E2E_PROVISION_FILE=/path/to/libvirt.properties CLOUD_PROVIDER=libvirt make test-e2e
+$ TEST_E2E_PROVISION=yes TEST_E2E_PROVISION_FILE=/path/to/libvirt.properties BUILTIN_CLOUD_PROVIDERS=libvirt make test-e2e
 ```
 
 The `TEST_E2E_PODVM_IMAGE` is an optional variable which specifies the path to the podvm qcow2 image. If it is set then the image should be uploaded to the VPC storage. The following command, as an example, instructs the tool to upload `path/to/podvm-base.qcow2` after the provisioning of the test environment:
 
 ```
-$ TEST_E2E_PROVISION=yes TEST_E2E_PODVM_IMAGE="path/to/podvm-base.qcow2" CLOUD_PROVIDER=libvirt make test-e2e
+$ TEST_E2E_PROVISION=yes TEST_E2E_PODVM_IMAGE="path/to/podvm-base.qcow2" BUILTIN_CLOUD_PROVIDERS=libvirt make test-e2e
 ```
 
 By default it is given 20 minutes for the entire e2e execution to complete, otherwise the process is preempted. If you need to extend that timeout then export the `TEST_E2E_TIMEOUT` variable. For example, `TEST_E2E_TIMEOUT=30m` set the timeout to 30 minutes. See `-timeout` in [go test flags](https://pkg.go.dev/cmd/go#hdr-Testing_flags) for the values accepted.

